presto-main/src/main/java/com/facebook/presto/sql/gen/PageFunctionCompiler.java

```diff
@@ -358,27 +369,35 @@ private MethodDefinition generateEvaluateMethod(
         return method;
     }
 
-    public Supplier<PageFilter> compileFilter(SqlFunctionProperties sqlFunctionProperties, RowExpression filter, Optional<String> classNameSuffix)
+    public Supplier<PageFilter> compileFilter(SqlFunctionProperties sqlFunctionProperties, RowExpression expression, Optional<String> classNameSuffix, boolean isBlockLargeByteCodeGeneration)
     {
         if (filterCache == null) {
-            return compileFilterInternal(sqlFunctionProperties, filter, classNameSuffix);
+            return compileFilterInternal(sqlFunctionProperties, expression, classNameSuffix, isBlockLargeByteCodeGeneration);
         }
-        return filterCache.getUnchecked(new CacheKey(sqlFunctionProperties, filter));
+        return filterCache.getUnchecked(new CacheKey(sqlFunctionProperties, expression, isBlockLargeByteCodeGeneration));
     }
 
-    private Supplier<PageFilter> compileFilterInternal(SqlFunctionProperties sqlFunctionProperties, RowExpression filter, Optional<String> classNameSuffix)
+    public Supplier<PageFilter> compileFilter(SqlFunctionProperties sqlFunctionProperties, RowExpression filter, Optional<String> classNameSuffix)
```

Comment:
why is there a version of compileFilter that doesn't take the isblockLargeBytecodeGeneration.  Don't we always want to do that check?   Also, we usually would put this version above the other one that it calls.

Reply:
Cruft. Removed. This whole passing flags down the call chain is super-annoying